### PR TITLE
code-gen-react-ui

### PR DESCRIPTION
**Pull request by jaswanth8888**
## Description

Added Code Gen React UI

## Issues

n/a.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies
- mantine/core
 - mantine/hooks
 - microsoft/fetch-event-source
 - reduxjs/toolkit
 - tabler/icons-react
 - axios
 - react
 - react-dom
 - react-markdown
 - react-syntax-highlighter
 - remark-frontmatter
 - remark-gfm

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: .github/workflows/scripts/build_push.sh [hunk 1]
```diff
@@ -54,6 +54,9 @@ for MEGA_SVC in $1; do
             docker_build ${IMAGE_NAME}-ui docker/Dockerfile
             if [ "$MEGA_SVC" == "ChatQnA" ];then
                 docker_build ${IMAGE_NAME}-conversation-ui docker/Dockerfile.react
+            fi
+             if [ "$MEGA_SVC" == "CodeGen" ];then
```
Comment (chensuyue) [inline .github/workflows/scripts/build_push.sh:58]: fix format
image

**Comment by chensuyue** [comment]
Add one line after this one to replace the registry, https://github.com/opea-project/GenAIExamples/blob/615f0d25470624534c541161c6e647f78b448af1/CodeGen/tests/test_codegen_on_xeon.sh#L43

**Comment by jaswanth8888** [comment]
> sed -i "s#image: opea/codegen-ui:latest#image: opea/codegen-ui:${IMAGE_TAG}#g" docker_compose.yaml

added

File: .github/workflows/scripts/build_push.sh (hunk 1 above)
Comment (jaswanth8888) [reply .github/workflows/scripts/build_push.sh:58]: fixed it

**Comment by chensuyue** [comment]
`docker compose -f docker_compose.yaml up -d` failed with https://github.com/opea-project/GenAIExamples/actions/runs/10034513589/job/27729334956?pr=417#step:5:554 , does it work in your local test? 
```
codegen-xeon-ui-server Pulled 
failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3891912248/Dockerfile: no such file or directory
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 2B done
#2 DONE 0.0s
```

**Comment by chensuyue** [comment]
> `docker compose -f docker_compose.yaml up -d` failed with https://github.com/opea-project/GenAIExamples/actions/runs/10034513589/job/27729334956?pr=417#step:5:554 , does it work in your local test?
> 
> ```
> codegen-xeon-ui-server Pulled 
> failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3891912248/Dockerfile: no such file or directory
> #1 [internal] load .dockerignore
> #1 transferring context: 2B done
> #1 DONE 0.0s
> 
> #2 [internal] load build definition from Dockerfile
> #2 transferring dockerfile: 2B done
> #2 DONE 0.0s
> ```

Seems the issues is it's not going to pull the docker image from our test registry but build from source, then not able to get the Dockerfile, if that's the issue, you should able reproduce in your local test.

**Comment by chensuyue** [comment]
> > `docker compose -f docker_compose.yaml up -d` failed with https://github.com/opea-project/GenAIExamples/actions/runs/10034513589/job/27729334956?pr=417#step:5:554 , does it work in your local test?
> > ```
> > codegen-xeon-ui-server Pulled 
> > failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3891912248/Dockerfile: no such file or directory
> > #1 [internal] load .dockerignore
> > #1 transferring context: 2B done
> > #1 DONE 0.0s
> > 
> > #2 [internal] load build definition from Dockerfile
> > #2 transferring dockerfile: 2B done
> > #2 DONE 0.0s
> > ```
> 
> Seems the issues is it's not going to pull the docker image from our test registry but build from source, then not able to get the Dockerfile, if that's the issue, you should able reproduce in your local test.

If this works in your local test, I will do more test to identify the issue.

**Comment by yogeshmpandey** [comment]
> > > `docker compose -f docker_compose.yaml up -d` failed with https://github.com/opea-project/GenAIExamples/actions/runs/10034513589/job/27729334956?pr=417#step:5:554 , does it work in your local test?
> > > ```
> > > codegen-xeon-ui-server Pulled 
> > > failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3891912248/Dockerfile: no such file or directory
> > > #1 [internal] load .dockerignore
> > > #1 transferring context: 2B done
> > > #1 DONE 0.0s
> > > 
> > > #2 [internal] load build definition from Dockerfile
> > > #2 transferring dockerfile: 2B done
> > > #2 DONE 0.0s
> > > ```
> > 
> > 
> > Seems the issues is it's not going to pull the docker image from our test registry but build from source, then not able to get the Dockerfile, if that's the issue, you should able reproduce in your local test.
> 
> If this works in your local test, I will do more test to identify the issue.

@chensuyue 

Tests works locally on Xeon. 

image

File: CodeGen/docker/ui/react/README.md
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/README.md:1]: removed and replaced with VITE_CODE_GEN_URL

File: CodeGen/docker/ui/react/public/vite.svg
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/public/vite.svg:1]: deleted the file

File: CodeGen/docker/ui/react/src/components/Message/conversationMessage.tsx
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/src/components/Message/conversationMessage.tsx:1]: removed the commented code

File: CodeGen/docker/ui/react/src/vite-env.d.ts
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/src/vite-env.d.ts:1]: This file is particular for the editors to recognise the environent variable available

Ref: https://vitejs.dev/guide/env-and-mode.html#intellisense-for-typescript

this is a template file added by VITE.

File: CodeGen/docker/xeon/docker_compose.yaml
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/xeon/docker_compose.yaml:1]: I have removed environment variables as we pass the BACKEND_SERVICE_ENDPOINT as build argument.

File: CodeGen/docker/gaudi/docker_compose.yaml
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply CodeGen/docker/gaudi/docker_compose.yaml:1]: Removed environment variables for xeon as well

File: CodeGen/tests/test_codegen_on_xeon.sh [hunk 2]
```diff
@@ -41,6 +42,7 @@ function start_services() {
         echo "using image repository $IMAGE_REPO and image tag $IMAGE_TAG"
         sed -i "s#image: opea/codegen:latest#image: opea/codegen:${IMAGE_TAG}#g" docker_compose.yaml
         sed -i "s#image: opea/codegen-ui:latest#image: opea/codegen-ui:${IMAGE_TAG}#g" docker_compose.yaml
+        sed -i "s#image: opea/codegen-react-ui:latest#image: opea/codegen-react-ui:${IMAGE_TAG}#g" docker_compose.yaml
```
Comment (chensuyue) [inline CodeGen/tests/test_codegen_on_xeon.sh:45]: Please add the same line for `CodeGen/tests/test_codegen_on_gaudi.sh`

**Review by yogeshmpandey — APPROVED** [review]

File: CodeGen/tests/test_codegen_on_xeon.sh (hunk 2 above)
Comment (jaswanth8888) [reply CodeGen/tests/test_codegen_on_xeon.sh:45]: added `CodeGen/tests/test_codegen_on_gaudi.sh` file

File: CodeGen/tests/test_codegen_on_gaudi.sh [hunk 3]
```diff
@@ -51,7 +53,7 @@ function start_services() {
     # Start Docker Containers
     docker compose -f docker_compose.yaml up -d
 
-    sleep 2m # Waits 2 minutes
+    sleep 5m # Waits 5 minutes
```
Comment (chensuyue) [inline CodeGen/tests/test_codegen_on_gaudi.sh:56]: I don't think you need to expand the sleep time.

File: CodeGen/tests/test_codegen_on_xeon.sh [hunk 4]
```diff
@@ -20,6 +20,7 @@ function build_docker_images() {
 
     cd $WORKPATH/docker/ui
     docker build --no-cache -t opea/codegen-ui:latest -f docker/Dockerfile .
+    docker build --no-cache --build-arg BACKEND_SERVICE_ENDPOINT=http://${ip_address}:7778/v1/codegen -t opea/codegen--react-ui:latest -f docker/Dockerfile.react .
```
Comment (chensuyue) [inline CodeGen/tests/test_codegen_on_xeon.sh:23]: fix the typo: `opea/codegen--react-ui:latest`

File: CodeGen/docker/gaudi/docker_compose.yaml [hunk 5]
```diff
@@ -72,6 +72,19 @@ services:
...
     restart: always
 
+  codegen-gaudi-react-ui-server:
+    image: opea/codegen-react-ui:latest
+    container_name: codegen-gaudi-react-ui-server
+    depends_on:
+      - codegen-gaudi-backend-server
+    ports:
+      - "5174:80"
+    build:
```
Comment (chensuyue) [inline CodeGen/docker/gaudi/docker_compose.yaml:82]: suggest to remove the build part.

Comment (chensuyue) [reply CodeGen/docker/gaudi/docker_compose.yaml:82]: found some random issues in CI,

File: CodeGen/docker/xeon/docker_compose.yaml [hunk 6]
```diff
@@ -66,6 +66,18 @@ services:
...
     ipc: host
     restart: always
+  codegen-xeon-react-ui-server:
+    image: opea/codegen-react-ui:latest
+    container_name: codegen-xeon-react-ui-server
+    depends_on:
+      - codegen-xeon-backend-server
+    ports:
+      - "5174:80"
+    build:
```
Comment (chensuyue) [inline CodeGen/docker/xeon/docker_compose.yaml:76]: suggest to remove the build part.

Comment (jaswanth8888) [reply CodeGen/docker/xeon/docker_compose.yaml:76]: build part is needed as we cannot inject environmental values to react once the image is built

File: CodeGen/docker/gaudi/docker_compose.yaml (hunk 5 above)
Comment (jaswanth8888) [reply CodeGen/docker/gaudi/docker_compose.yaml:82]: build part is need as we cannot inject environmental values to react once the image is built.
Can't the see the image that you have shared.

File: CodeGen/tests/test_codegen_on_gaudi.sh (hunk 3 above)
Comment (jaswanth8888) [reply CodeGen/tests/test_codegen_on_gaudi.sh:56]: reverted back to 2m

File: CodeGen/tests/test_codegen_on_xeon.sh (hunk 4 above)
Comment (jaswanth8888) [reply CodeGen/tests/test_codegen_on_xeon.sh]: fixed typo

File: CodeGen/docker/ui/react/src/styles/components/_context.scss
Comment (WenjiaoYue) [inline CodeGen/docker/ui/react/src/styles/components/_context.scss:1]: Should empty files be deleted?

File: CodeGen/docker/ui/react/src/assets/react.svg
Comment (WenjiaoYue) [inline CodeGen/docker/ui/react/src/assets/react.svg:1]: Should useless file be deleted?

File: CodeGen/docker/ui/react/src/styles/components/_context.scss
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/src/styles/components/_context.scss:1]: deleted

File: CodeGen/docker/ui/react/src/assets/react.svg
Comment (jaswanth8888) [reply CodeGen/docker/ui/react/src/assets/react.svg:1]: deleted

**Review by WenjiaoYue — APPROVED** [review]

**Pull request merged**